libraries/psio/include/psio/member_proxy.hpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
```

Comment:
We should remove `member_proxy`. It doesn't seem to serve any useful purpose and it creates additional possibility for error:
```
auto x = view->m1()->m2();
// BOOM: x is a member_proxy that holds a reference to the temprorary returned by m1.
```